launchpad-guaranteed-tickets-v2/src/token_release.rs

```diff
@@ -93,6 +92,55 @@ pub trait TokenReleaseModule:
         self.emit_set_unlock_schedule_event(milestones);
     }
 
+    #[payable("*")]
+    #[only_owner]
+    #[endpoint(refundWinningTickets)]
+    fn refund_winning_tickets(&self, users: MultiValueEncoded<ManagedAddress>) {
+        let (payment_token, payment_amount) = self.call_value().egld_or_single_fungible_esdt();
+
+        let launchpad_tokens_per_winning_ticket = self.launchpad_tokens_per_winning_ticket().get();
+        let ticket_price = self.ticket_price().get();
+
+        require!(
+            payment_token == ticket_price.token_id,
+            "Invalid payment token"
+        );
+
+        let mut remaining_payment_amount = payment_amount;
+        for user in users {
+            let user_claimed_balance = self.user_claimed_balance(&user).get();
+            require!(
+                user_claimed_balance == BigUint::zero(),
+                "User already claimed tokens"
```

Comment:
So if one user has `claimed_balance != 0`, you just drop the execution of the entire endpoint?
Why not continue to the next user?

Reply:
Because I think it's better for the caller to know that the output of the endpoint had some unexpected outcomes. This operation should work only under special conditions, so there should not be any room for interpretation.